docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,921 @@
+| `UpToDateReplicas` (new)         | `UpToDateReplicas`                                       | `UpToDateReplicas`                  |
+| other fields...                  | other fields...                                          | other fields...                     |
+
+Notes:
+- The `BackCompatibilty` struct is going to exist in v1beta2 types only until v1beta1 removal (9 months or 3 minor releases after v1beta2 is released/v1beta1 is deprecated, whichever is longer).
+  Fields in this struct are used for supporting down conversions, thus providing users relying on v1beta1 APIs additional buffer time to pick up the new changes.
+- This proposal is using `UpToDateReplicas` instead of `UpdatedReplicas`; This is a deliberated choice to avoid 
+  confusion between update (any change) and upgrade (change of the Kubernetes versions).
+- Also `AvailableReplicas` will determine Machine's availability by reading Machine.Available condition instead of
+  computing availability as of today, however in this case the semantic of the field is not changed
+
+TODO: check `FullyLabeledReplicas`, do we still need it?
+
+#### MachineSet (New)Conditions
+
+| Condition        | Note                                                                                                           |
+|------------------|----------------------------------------------------------------------------------------------------------------|
+| `ReplicaFailure` | This condition surfaces issues on creating Machines, if any.                                                   |
+| `MachinesReady`  | This condition surfaces detail of issues on the controlled machines, if any.                                   |
+| `ScalingUp`      | True if available replicas < desired replicas                                                                  |
+| `ScalingDown`    | True if replicas > desired replicas                                                                            |
+| `UpToDate`       | True if all the Machines controlled by this MachineSet are up to date (replicas = upToDate replicas)           |
+| `Remediating`    | True if there is at least one Machine controlled by this MachineSet is not passing health checks               |
+| `Deleted`        | True if MachineSet is deleted; Reason can be used to observe the cleanup progress when the resource is deleted |
+| `Paused`         | True if this MachineSet or the Cluster it belongs to are paused                                                |
+
+> To better evaluate proposed changes, below you can find the list of current MachineSet's conditions:
+> Ready, MachinesCreated, Resized, MachinesReady.
+
+Notes:
+- MachineSet conditions are intentionally mostly consistent with MachineDeployment conditions to help users troubleshooting .
+- MachineSet is considered as a sort of implementation detail of MachineDeployments, so it doesn't have its own concept of availability.
```

Comment:
If I look at prior art, the same assumption was used for ReplicaSet in K8s.

I think this is also the case for CAPI, because also MachineSet do not have any sort of "not aivaliabiltiy" they tolerate, and thus it does not make sense to talk about availabity too